onnxscript/tests/function_libs/torch_lib/ops_test.py

```diff
@@ -42,16 +42,20 @@ def dtypes_except(*dtypes: torch.dtype) -> Sequence[torch.dtype]:
     return tuple(dtype for dtype in TESTED_DTYPES if dtype not in dtypes)
 
 
-def _should_skip_test_sample(op_name: str, sample) -> Optional[str]:
+def _should_skip_xfail_test_sample(op_name: str, sample) -> Optional[Tuple[str, Callable]]:
     """Returns a reason if a test sample should be skipped."""
-    if op_name not in ops_test_data.OP_WITH_SKIPPED_SUBTESTS:
+    if op_name not in ops_test_data.OP_WITH_SKIPPED_XFAIL_SUBTESTS:
         return None
-    for decorator_meta in ops_test_data.SKIP_SUBTESTS:
-        # Linear search on ops_test_data.SKIP_SUBTESTS. That's fine because the list is small.
+    for decorator_meta in ops_test_data.SKIP_XFAIL_SUBTESTS:
+        # Linear search on ops_test_data.SKIP_XFAIL_SUBTESTS. That's fine because the list is small.
         if decorator_meta.op_name == op_name:
             assert decorator_meta.matcher is not None, "Matcher must be defined"
             if decorator_meta.matcher(sample):
-                return decorator_meta.reason
+                assert decorator_meta.name in ("skip", "xfail"), "Invalid decorator_meta name"
+                if decorator_meta.name == "xfail":
+                    return decorator_meta.reason, pytest.xfail
+                else:
```

Comment:
Possible to return the decorator field instead?

Reply:
Leave the if/else condition outside this function?